src/Slider/MultiLayerSlider/MultiLayerSlider.jsx

```diff
@@ -0,0 +1,166 @@
+  }
+
+  /**
+   * Formats the tip for the slider.
+   * @param  {Number} value the slider value
+   * @return {String}      the formatted tip value
+   */
+  formatTip(value) {
+    const layerIdx = this.getLayerIndexForSliderValue(value);
+    const layers = this.props.layers;
+    let tip;
+    if (layers[layerIdx]) {
+      const opacity = Math.round(layers[layerIdx].get('opacity') * 100);
+      const layer = layers[layerIdx];
+      const layername = layer.get('name') || layer.get('title');
+      tip = `${layername} ${opacity}%`;
+    }
+    return tip;
+  }
+
+  /**
+   * Called when the value of the slider changed.
+   * @param  {Number} value the new value
+   */
+  valueUpdated(value) {
+    const layerIdx = this.getLayerIndexForSliderValue(value);
+    const opacity = this.getOpacityForValue(value);
+    const layers = this.props.layers;
+    // set all opacities to 0 first
+    layers.forEach(l => l.setOpacity(0));
+    if (layers[layerIdx]) {
+      layers[layerIdx].setOpacity(1 - opacity);
+    }
+    if (layers[layerIdx - 1] && opacity > 0.5) {
+      layers[layerIdx - 1].setOpacity(opacity - 0.5);
+    }
+    if (layers[layerIdx + 1]) {
+      layers[layerIdx + 1].setOpacity(opacity);
+    }
+  }
+
+  /**
+   * Gets the opacity for a given slider value
+   * @param  {Number} value The current slider value
+   * @return {Number} The opacity
+   */
+  getOpacityForValue(value) {
+    const length = this.props.layers.length - 1;
+    const ticksPerLayer = Math.round(100 / length);
+    const idx = parseInt(value / ticksPerLayer, 10);
+    const opacity = value / ticksPerLayer - (idx > length ? length : idx);
+    return opacity > 1 ? 1 : opacity;
+  }
+
+  /**
+   * Gets the matching index of the layer array for a given slider value
+   * @param  {Number} value the current slider value
+   * @return {Number} The layer array index
+   */
+  getLayerIndexForSliderValue(value) {
+    const length = this.props.layers.length - 1;
+    const ticksPerLayer = Math.round(100 / length);
+    const idx = parseInt(value / ticksPerLayer, 10);
+    return idx > length ? length : idx;
+  }
+
+  /**
+   * Creates the marks used with the slider based on the layers names.
+   * @return {Object} The marks object
+   */
+  getMarks() {
+    const marks = {};
+    const layers = this.props.layers;
+    const length = layers.length - 1;
+    layers.forEach(function(l, i) {
+      const layername = l.get('name') || l.get('title') || 'Layer ' + i + 1;
+      const idx = Math.round(100 / length * i);
+      marks[idx] = layername;
+    });
+    return marks;
+  }
+
+  /**
+   * The render function.
+   */
+  render() {
+    const {
+      layers,
+      defaultValue,
+      className,
+      ...passThroughProps
+    } = this.props;
+
+    const finalClassName = className
+      ? `${className} ${this.className}`
+      : this.className;
+
+    return (
+      <Slider
+        className={finalClassName}
+        marks={this.getMarks()}
```

Comment:
shouldn't the `bind` function be used here (like `this.getMarks().bind(this)`)?

Reply:
Nice catch.

Reply:
No bind needed here, this breaks functionality as there is no function expected. Will be fixed in upcoming PR